content/tutorials/aws-persistence.textile

```diff
@@ -0,0 +1,255 @@
+# Click @Create function@
+
+You'll be presented with the function page with an empty "lambda_function.py" in your page:
+
+<a href="/images/tutorials/persistence/new-lambda.png" target="_blank">
+  <img src="/images/tutorials/persistence/new-lambda.png" style="width: 100%" alt="New lambda page">
+</a>
+
+Update @lambda_function.py@ with the following code, remembering to replace the bucket name with the one you created:
+
+```[python]
+import json, boto3, os, sys, uuid
+from urllib.parse import unquote_plus
+
+s3_client = boto3.client('s3')
+
+def lambda_handler(event, context):
+    
+    event_name = (event['name'])                   # Choosing name field
+    message = (event['data'])                      # Choosing data field
+    data_to_save = event_name + ' : ' + message    # Combining fields
+    print(data_to_save)                            # Checking combination
+    
+    bucket_name = "your-bucket-name-here"          # Target S3 Bucket
+    file_name = "test.txt"                         # Target file
+    lambda_path = "/tmp/" + file_name              # Path of file being download
+    s3_path = "output/" + file_name                # Path of file being uploaded
+    
+    file_object = open(lambda_path, 'a')           # Open downloaded file to append
+    file_object.write("\n" + data_to_save)         # Update downloaded file
+    file_object.close()                            # Save and close edited file
+    
+    s3 = boto3.resource("s3")
+    s3.meta.client.upload_file(lambda_path, bucket_name, file_name)
+```
+
+Once the webhook is set up this code will be invoked every time a message is published. When invoked it will take the data from the JSON Ably sends to Lambda, then open our @test.txt@, append the latest message, save the message and finally upload it back to S3.
+
+Now you're finished in AWS you can set up the Ably half of the workflow.
+
+h2(#installation). Set up Ably app and API Key
+
+You're going to need an API key to use when sending messages to our channel. In your "Ably dashboard":https://ably.com/accounts:
+
+# Hover over @Dashboard@ then select @Apps Dashboard@
+# Click @Create New App@
+# Name your new app
+# Copy the API Key somewhere safe
+
+h2(#webhook). Set up the webhook 
+
+Now you have an app you can set up an integration rule with AWS Lambda to trigger when a new message is published using the app. In your "Ably dashboard":https://ably.com/accounts:
+
+# Hover over @Dashboard@ then select @Apps Dashboard@
+# Click the name of your app in the URL breadcrumb 
+# Click @Integrations@ then @New Integration Rule@
+# Select @Webhook@ by clicking @Choose@ below it then 
+# Select @AWS Lambda@ by clicking @Choose@ below it
+# Set your webhook settings:
+## Set @AWS Region@ to the the Region your AWS account is located in
+## Enter @ably-persistence@ as the @Function Name@
+## Enter the AWS Key you saved earlier as your @AWS Credentials@
```

Comment:
We should be recommending using an IAM role rather than user credentials. Perhaps this should just link to the AWS Authentication doc? https://ably.com/docs/general/aws-authentication